server/src/main/java/org/opensearch/plugins/StreamManagerPlugin.java

```diff
@@ -23,5 +25,13 @@ public interface StreamManagerPlugin {
      *
      * @return The StreamManager instance
      */
-    Supplier<StreamManager> getStreamManager();
+    default Supplier<StreamManager> getStreamManager() {
+        return null;
+    }
+
+    /**
+     * Called when the StreamManager is initialized.
+     * @param streamManager Supplier of the StreamManager instance
+     */
+    default void onStreamManagerInitialized(Supplier<StreamManager> streamManager) {}
```

Comment:
So what is the reason to pass `Supplier<StreamManager>` here? The plugins will call `Supplier::get` multiple times, causing multiple initialization, right? Shouldn't we pass fully initialized instance of `StreamManager ` instead?
```suggestion
    default void onStreamManagerInitialized(StreamManager streamManager) {}
```

Reply:
removed supplier altogether, we probably don't need it. I had some requirements earlier but not anymore.